src/Adapter/Product/Combination/QueryHandler/GetEditableCombinationsListHandler.php

```diff
@@ -122,27 +142,33 @@ public function handle(GetEditableCombinationsList $query): CombinationListForEd
             $query->getLanguageId()
         );
 
+        $imageIdsByCombinationIds = $this->productImageRepository->getImagesIdsForCombinations($combinationIds);
+
         return $this->formatEditableCombinationsForListing(
             $combinations,
             $attributesInformation,
-            $total
+            $total,
+            $imageIdsByCombinationIds
         );
     }
 
     /**
      * @param array $combinations
      * @param array<int, array<int, mixed>> $attributesInformationByCombinationId
      * @param int $totalCombinationsCount
+     * @param array $imageIdsByCombinationIds
      *
      * @return CombinationListForEditing
      */
     private function formatEditableCombinationsForListing(
         array $combinations,
         array $attributesInformationByCombinationId,
-        int $totalCombinationsCount
+        int $totalCombinationsCount,
+        array $imageIdsByCombinationIds
     ): CombinationListForEditing {
         $combinationsForEditing = [];
 
+        $cachedImages = [];
```

Comment:
Please add a property to the class  rather than a reference object

Reply:
Agreeds using `&$cachedImages` as a parameter is a bit ugly, you can cache the same way by using a class field